pkg/services/object/put/distributed.go

```diff
@@ -187,12 +190,34 @@ func (t *distributedTarget) Close() (oid.ID, error) {
 			return id, nil
 		}
 
-		err = t.cnrClient.SubmitObjectPut(await, t.objSharedMeta, t.collectedSignatures)
+		addr := object.AddressOf(t.obj)
+		var objAccepted chan error
+		if await {
+			objAccepted = make(chan error, 1)
+			t.metaSvc.NotifyObjectSuccess(objAccepted, addr)
+		}
+
+		err = t.cnrClient.SubmitObjectPut(t.objSharedMeta, t.collectedSignatures)
 		if err != nil {
-			return oid.ID{}, fmt.Errorf("failed to submit %s object meta information: %w", id, err)
+			if await {
+				t.metaSvc.UnsubscribeFromObject(addr)
+			}
+			return oid.ID{}, fmt.Errorf("failed to submit %s object meta information: %w", addr, err)
+		}
+
+		if await {
+			select {
+			case <-time.After(15 * time.Second):
+				t.metaSvc.UnsubscribeFromObject(addr)
```

Comment:
this is done in 2 cases, i think it'd be easier to just `defer` right after `NotifyObjectSuccess`. Correct though

Reply:
well, i wanted to do so, but in a success case (most of the expected cases) it will be an additional sync lock for nothing